tiledb/sm/misc/CMakeLists.txt

```diff
@@ -156,13 +156,14 @@ if (TILEDB_TESTS)
   target_link_libraries(unit_misc PRIVATE math)
   find_package(Catch_EP REQUIRED)
   target_link_libraries(unit_misc PUBLIC Catch2::Catch2WithMain)
+  target_link_libraries(unit_misc PUBLIC global_state storage_manager_stub uuid)
```

Comment:
`storage_manager_stub` is OK here, since it's a dependency of a unit test, but nonetheless it's suspicious. Is it needed here? If so, why?